scripts/linters/js_ts_linter.py

```diff
@@ -901,6 +901,104 @@ def _check_constants_declaration(self):
+        in the comment.
+        """
+        if self.verbose_mode_enabled:
+            python_utils.PRINT(
+                'Starting comment checks\n'
+                '----------------------------------------')
+        summary_messages = []
+        files_to_check = self.all_filepaths
+        allowed_terminating_punctuations = [
+            '.', '?', ';', ',', '{', '^', ')', '}', '>']
+
+        # We allow comments to not have a terminating punctuation if any of the
+        # below phrases appears at the beginning of the comment.
+        # Example: // eslint-disable max-len
+        # This comment will be excluded from this check.
+        allowed_start_phrases = [
+            '@ts-ignore', '--params', 'eslint-disable', 'eslint-enable']
+
+        # We allow comments to not have a terminating punctuation if any of the
+        # below phrases appears in the last word of a comment.
+        # Example: // Ref: https://some.link.com
+        # This comment will be excluded from this check.
+        allowed_end_phrases = ['http://', 'https://']
+
+        failed = False
+        with linter_utils.redirect_stdout(sys.stdout):
+            for filepath in files_to_check:
+                file_content = FILE_CACHE.readlines(filepath)
+                file_length = len(file_content)
+                for line_num in python_utils.RANGE(file_length):
+                    line = file_content[line_num].strip()
+                    next_line = ''
+                    previous_line = ''
+                    if line_num + 1 < file_length:
+                        next_line = file_content[line_num + 1].strip()
+
+                    # Exclude comment line containing heading.
+                    # Example: // ---- Heading ----
+                    # These types of comments will be excluded from this check.
+                    if (
+                            line.startswith('//') and line.endswith('-')
+                            and not (
+                                next_line.startswith('//') and
+                                previous_line.startswith('//'))):
+                        continue
+
+                    if line.startswith('//') and not next_line.startswith('//'):
+                        # Check if any of the allowed starting phrase is present
+                        # in comment and exclude that line from check.
+                        allowed_start_phrase_present = any(
+                            line.split()[1].startswith(word) for word in
+                            allowed_start_phrases)
+
+                        if allowed_start_phrase_present:
+                            continue
+
+                        # Check if any of the allowed ending phrase is present
+                        # in comment and exclude that line from check.
+                        allowed_end_phrase_present = any(
+                            word in line.split()[-1] for word in
+                            allowed_end_phrases)
```

Comment:
I used `in` here instead of `startswith` because some urls have quotes around them

Example: // Ref: 'https://oppia.org'

Reply:
Done. Added an inline comment.